README.md

```diff
@@ -319,6 +319,18 @@ role.
 Similar to `controller` and `vlan`, the `parent` references the connection profile in
 the ansible role.
 
+#### Network_connections Module
+
+The `network_connections.py module` is intended for internal usage by the role and It is not meant to be directly accessed or utilised by external roles or playbooks.Contains the internal Ansible module, which is the main script. It controls the communication between the role and Ansible, imports the YAML configuration and applies the changes to the provider (i.e. NetworkManager, initscripts).
+
+### Purpose of  Network_connections Module
+The primary purpose of the `network_connections module` is to provide internal utility functions and classes for managing network connections and related tasks within the context of the role.
```

Comment:
If you run the `tests_reapply_nm.yml`, you will notice that the internal module can be used to speed up the tests running because the tasks that are specified in `tasks/main/yml` are skipped.